gridcomps/History/Sampler/MAPL_StationSamplerMod.F90

```diff
@@ -73,155 +101,226 @@ function new_StationSampler_readfile (filename,nskip_line, rc) result(sampler)
-
-    rewind(unit)
-    if (nskip>0) then
-       do i=1, nskip
-          read(unit, *)
+       sampler%nstation=nstation
+       allocate(sampler%station_id(nstation), _STAT)
+       allocate(sampler%station_name(nstation), _STAT)
+       allocate(sampler%station_fullname(nstation), _STAT)
+       allocate(sampler%lons(nstation), _STAT)
+       allocate(sampler%lats(nstation), _STAT)
+       allocate(sampler%elevs(nstation), _STAT)
+
+       rewind(unit)
+       if (nskip>0) then
+          do i=1, nskip
+             read(unit, *)
+          end do
+       end if
+       do i=1, nstation
+          if(seq=='IAFFF') then
+             read(unit, *) &
+                  sampler%station_id(i), &
+                  sampler%station_name(i), &
+                  sampler%lons(i), &
+                  sampler%lats(i)
+          elseif(seq=='AIFFF') then
+             read(unit, *) &
+                  sampler%station_name(i), &
+                  sampler%station_id(i), &
+                  sampler%lons(i), &
+                  sampler%lats(i)
+          elseif(trim(seq)=='AFF' .OR. trim(seq)=='AFFF') then
+             !!write(6,*) 'i=', i
+             line=''
+             read(unit, '(a100)')  line
+             !!write(6,*) 'line=', trim(line)
+             call CSV_read_line_with_CH_I_R(line, &
+                  sampler%station_name(i), &
+                  sampler%lons(i), &
+                  sampler%lats(i), _RC)
+             sampler%station_id(i)=i
+          elseif(trim(seq)=='AFFFA') then
+          ! NOAA GHCNd
+          ! Ex: 'CHM00054511  39.9330  116.2830   55.0    BEIJING   GSN     54511'
+             read(unit, *) &
+                  sampler%station_name(i), &
+                  sampler%lats(i), &
+                  sampler%lons(i)
+             sampler%station_id(i)=i
+             backspace(unit)
+             read(unit, '(a100)', IOSTAT=ios) line
+             j=index(line, '.', BACK=.true.)
+             line2=line(j+1:)
+             k=len(line2)
+             line=''
+             do j=1, k
+                CH1=line2(j:j)
+                con1= (CH1>='a'.AND.CH1<='z').OR.(CH1>='A'.AND.CH1<='Z')
+                if (con1) exit
+             enddo
+             read(line2(j:k), '(a100)') line
+             line2=trim(line)
+             k=len(line2)
+             line=''
+             do j=1, k
+                CH1=line2(j:j)
+                con1= (CH1>='0' .AND. CH1<='9')
+                if (con1) exit
+             enddo
+             if (j>k) j=k
+             sampler%station_fullname(i) = trim(line2(1:j-1))
+          end if
        end do
+       close(unit)
+
+       write(6,*)  'nstation=', nstation
+       write(6,*)  'sampler%station_name(1:2) : ', &
+            trim(sampler%station_name(1)), trim(sampler%station_name(2))
+       write(6,*)  'sampler%lons(1:2) : ',&
+            sampler%lons(1:2)
+       write(6,*)  'sampler%lats(1:2) : ',&
+            sampler%lats(1:2)
```

Comment:
This should be done with pflogger.   I'll approve as is, but please follow up with another PR to change this.

Reply:
It is in the `if ( MAPL_AM_I_ROOT() )` block and flogger did not work when I tried.